core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -328,6 +328,14 @@ public int numConnections() {
         return connectionLimitingHandler.numConnections();
     }
 
+    /**
+     * Returns the result of {@link CompletableFuture} which is completed after the {@link #close()} or
+     * {@link #closeAsync()} operation is completed.
+     */
+    public Object blockUntilShutdown() throws Exception {
```

Comment:
I think we don't need to return an object here. How about making the return type `void`?

Reply:
It would be nice to add a test that registers a [ServerListener](https://line.github.io/armeria/apidocs/com/linecorp/armeria/server/ServerListener.html) to test server and could check `serverStopped(...)` was invoked.

Reply:
Hi,
I changed the method, but I'm having trouble with the test. I don't know how to check if server stop has been invoked. Can you help? Thanks!

Reply:
You could do this:

```java
assertThat(server.activePorts()).isEmpty();
```

Reply:
You can test the method like the following.  
```
@Test
void testShutDownServer() {
    final AtomicBoolean stopped = new AtomicBoolean();
    final Server server = Server.builder()
                                .service("/", (ctx, req) -> HttpResponse.of(HttpStatus.OK))
                                .serverListener(new ServerListenerAdapter() {
                                    @Override
                                    public void serverStopping(Server server) throws Exception {
                                        stopped.set(true);
                                    }
                                })
                                .build();
    server.start().join();
    CommonPools.blockingTaskExecutor().schedule(server::close, 1, TimeUnit.SECONDS);
    server.whenClosed().join(); // replace with new method
    assertThat(stopped).isTrue();
}
```

Reply:
I see.. I was going for a ServerTest and a ThreadRecordingServerListener as a listener, thanks!